plugins/videobasedtracker/UVCImageSource.cpp

```diff
@@ -173,12 +176,16 @@ namespace vbtracker {
 
     bool UVCImageSource::grab()
     {
-        // libuvc doesn't provide for separate grab() and retrieve() operations.
-        // Instead, grab() will set a flag to alert the callback function that
-        // we wish to store the next frame and retrieve() will return that
-        // frame.
-        std::lock_guard<std::mutex> guard(mutex_);
-        return (frame_ != nullptr);
+      //Gain access to frames_ queue
+      std::unique_lock<std::mutex> lock(mutex_);
+
+      //Check if there are any frames available
+      if (frames_.empty())
+	//Wait for some frames to become available
+	frames_available_.wait(lock);
```

Comment:
can you put curly braces around the "then" part here, and also run clang-format?

Also, I think this would be subject to spurious wakeups - might want to use the other form of `.wait` that lets you pass in a predicate.